src/core/monitors/VariableMonitor.h

```diff
@@ -4,19 +4,32 @@
 #include <cstddef>
 #include <vector>
 #include <iosfwd>
+#include "variant.h"
 
 #include "AbstractFileMonitor.h"
 #include "MonteCarloAnalysisOptions.h"
 
+struct SeparatorFormat
```

Comment:
For future consideration, is it conceivable that file writers/readers outside of VariableMonitor could make use SeparatorFormat? For example, the `write` method assumes csv/tsv format and accepts a separator flag, though it could probably be adapted to support json. If so, it may be worthwhile separating SeparatorFormat and JSONFormat out into their own file(s).

Reply:
Yeah, that might be a good idea.  In preparation for that, I move the file format types out into their own file `core/util/FileFormat.h`